integration/spark/src/main/java/openlineage/spark/agent/lifecycle/RddExecutionContext.java

```diff
@@ -170,15 +173,22 @@ public void start(SparkListenerJobStart jobStart) {
 
   @Override
   public void end(SparkListenerJobEnd jobEnd) {
-    LineageEvent event =
-        LineageEvent.builder()
+    OpenLineage ol = new OpenLineage(URI.create(OpenLineageClient.OPEN_LINEAGE_CLIENT_URI));
+    OpenLineage.RunEvent event =
+        ol.newRunEventBuilder()
+            .eventTime(toZonedTime(jobEnd.time()))
+            .eventType(getEventType(jobEnd.jobResult()))
             .inputs(buildInputs(inputs))
             .outputs(buildOutputs(outputs))
-            .run(buildRun(buildRunFacets(buildJobErrorFacet(jobEnd.jobResult()))))
+            .run(
+                ol.newRunBuilder()
+                    .runId(
+                        PlanUtils.convertToUUID
+                            .apply(sparkContext.getParentRunId())
+                            .orElse(UUID.randomUUID()))
```

Comment:
ok, so we should have runId also in context as parentRunId? We should discuss that logic

Reply:
should we then map jobId to some UUID and then set runId based on mapping jobId to UUID? Cause current logic is incorrect, I suppose

Reply:
Hmm, the `SparkSQLExecutionContext` impl is correct- the `RddExecutionContext` is wrong. 
See https://github.com/OpenLineage/OpenLineage/blob/main/integration/spark/src/main/java/openlineage/spark/agent/lifecycle/SparkSQLExecutionContext.java#L163 and https://github.com/OpenLineage/OpenLineage/blob/main/integration/spark/src/main/java/openlineage/spark/agent/lifecycle/SparkSQLExecutionContext.java#L105-L108 . Most notably, the `runUuid` is only created once and passed to both the start and the end event. Otherwise, the server would be receiving a start event for one run id and an end event for another run id and there's no way to know that they are actually from the same run.